app/services/save_practice_service.rb

```diff
@@ -196,16 +208,20 @@ def update_category_practices
       other_practice_categories = practice_categories.where(is_other: true)
 
       if other_practice_categories.any?
-        if cat_keys.exclude?(other_cat_id.to_s)
-          practice_category_practices.joins(:category).where(categories: {is_other: true}).destroy_all
-
-          other_practice_categories.each do |oc|
-            oc.destroy unless CategoryPractice.where.not(practice_id: @practice.id).where(category_id: oc.id).any?
+        other_parent_cat_options = ['other-clinical', 'other-operational', 'other-strategic']
+        other_parent_cat_options.each do |opc|
+          parent_cat = Category.where('lower(name) = ?', opc.split('-').pop).first
+          if cat_keys.exclude?(opc)
+            practice_category_practices.joins(:category).where(categories: { parent_category_id: parent_cat.id, is_other: true }).destroy_all
+
+            other_practice_categories.each do |oc|
+              oc.destroy unless oc.parent_category != parent_cat && CategoryPractice.where(category: oc).where.not(practice: @practice)
```

Comment:
We need to add `.present?` to the end of this `oc.destroy unless oc.parent_category != parent_cat && CategoryPractice.where(category: oc).where.not(practice: @practice)` since right now it deletes it

Reply:
added present?